modules/EventEmitter.js

```diff
@@ -0,0 +1,139 @@
+class EventEmitter {
+  listeners = {};  // key-value pair
+
+  _newEvent(eventName, fn) {
+    if (!this.listeners[`${eventName}`]) {
+      this.listeners[`${eventName}`] = [];
+    }
+    this.listeners[`${eventName}`].push({function: fn, once: false })
+  }
+
+  _deleteEvent(eventName, fn) {
+    const arrEvents = this.listeners[`${eventName}`];
+    for (let i = 0; i < arrEvents.length; i++) {
+      if (arrEvents[i].function === fn) {
+        arrEvents.splice(i, 1);
+      }
+    }
```

Comment:
Here you also can do a bit shorter:
```suggestion
    if (arrEvents) {
      this.listeners[`${eventName}`] = arrEvents.filter(event => event !== fn);
    }
```